src/app_service/service/message/service.nim

```diff
@@ -261,6 +270,16 @@ QtObject:
     self.events.emit(SIGNAL_RELOAD_MESSAGES, ReloadMessagesArgs(communityId: communityId))
 
   proc init*(self: Service) =
+    self.events.on(SignalType.MessageDelivered.event) do(e: Args):
+      let receivedData = MessageDeliveredSignal(e)
+      let data = MessageDeliveredArgs(chatId: receivedData.chatId, messageId: receivedData.messageId)
+      self.events.emit(SIGNAL_MESSAGE_DELIVERED, data)
```

Comment:
Do we store somewhere locally the message in a service? if yes, they should be updated too no?

Reply:
@alaibe I am very new to Nim and logic here - this is like second time I deal with it. So I cannot tell for sure about it. But just taking quick look at signals like SIGNAL_NEW_MESSAGE_RECEIVED or SIGNAL_NEW_MESSAGE_RECEIVED I dont see any service to store locally the message.